server.go

```diff
@@ -1011,6 +1011,7 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 		Clock:               clock.NewDefaultClock(),
 		StrictZombiePruning: strictPruning,
 		IsAlias:             aliasmgr.IsAlias,
+		HaveChannelWith:     s.htlcSwitch.HaveLinkWith,
```

Comment:
Is there a reason we switch from `Link` to `Channel` here? Otherwise it might cause confusion.